docs/usage/use-specification-dbcontext.md

```diff
@@ -8,5 +8,46 @@ nav_order: 3
 # How to use Specifications with a DbContext
 
 You can use Specifications to define queries that are executed directly using an EF6 or EF Core `DbContext`.
+The following snippet defines a `Customer` entity and a sample `DbContext` which defines a `DbSet` of Customers.
 
-(todo)
+```csharp
+public class Customer
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+}
+
+public class SampleDbContext : DbContext
+{
+    public DbSet<Customer> Customers { get; set; }
+
+    public SampleDbContext(DbContextOptions<SampleDbContext> options)
+        : base(options)
+    {
+    }
+
+    // additional overrides intentionally left out
+}
+```
+
+A specification can be applied directly to this `DbSet` using the `WithSpecification` extension method defined in `Ardalis.Specification.EntityFrameworkCore` package. Assuming a Specification is defined similar to the `ItemByIdSpec` described in [How to Create Specifications](./creating-specifications.md), the following code demonstrates putting these pieces together.
+
+```csharp
+int id = 1;
+
+var specification = new CustomerByIdSpec(id);
+
+var customer = dbContext.Customers
+    .WithSpecification(specification)
```

Comment:
Since this will only work if the required `using` statement is available can we make note of that or include it in the listing somehow? A comment at the top is probably sufficient.

Reply:
updated